### ISSUE TITLE
Sparse new-group prediction ignores supplied location and scale draws on dev

### ISSUE THREAD

**Issue by AlexanderFengler** [bambinos/bambi#1068]
## Summary

On the current `dev` branch, predictions for a previously unseen group label with `SPARSE_DOT=True` do not follow changes in the supplied draws of the group-location and scale hyperparameters. Dense prediction does. The setting should change the computation, not the conditioning of the predictive distribution.

This reproduces with centered priors on unmodified `dev` at `5b7e1b1f42097adeb1b9d846e30c31aa91616332`, without the changes in #1066. I have not established whether a released version is affected.

## Reproducer

The test uses PyMC's mock sampler only to obtain a small, correctly shaped trace. It then checks a deterministic property of prediction, not posterior convergence. Holding the prediction seed fixed, shifting both the stored coefficients and their population location by 3 should shift both known- and new-group predictions by 3.

```python
from unittest.mock import patch

import bambi as bmb
import numpy as np
import pandas as pd
from pymc.testing import mock_sample

data = pd.DataFrame({
    "y": [0., 1., 2., 3., 4., 5.],
    "g": ["a", "a", "b", "b", "c", "c"],
})
new_data = pd.DataFrame({"g": ["a", "new_group"]})
original_sparse = bmb.config.SPARSE_DOT
try:
    for sparse in (False, True):
        bmb.config.SPARSE_DOT = sparse
        model = bmb.Model(
            "y ~ 0 + (1|g)", data, noncentered=False,
            priors={"1|g": bmb.Prior(
                "Normal",
                mu=bmb.Prior("Normal", mu=2, sigma=1),
                sigma=bmb.Prior("HalfNormal", sigma=1),
            )},
        )
        with patch("pymc.sample", mock_sample):
            trace = model.fit(draws=10, chains=2, random_seed=12)

        before = model.predict(trace, data=new_data, random_seed=42, inplace=False)
        shifted = trace.copy(deep=True)
        shifted.posterior["1|g"] += 3
        shifted.posterior["1|g_mu"] += 3
        after = model.predict(shifted, data=new_data, random_seed=42, inplace=False)
        delta = (after.predictions["mu"] - before.predictions["mu"]).values
        print(sparse, np.unique(np.round(delta[..., 0], 8)),
              np.unique(np.round(delta[..., 1], 8)))
finally:
    bmb.config.SPARSE_DOT = original_sparse
```

Output (sparse flag, known-group shift, new-group shift):

```text
False [3.] [3.]
True  [3.] [0.]
```

Expected: both rows give `[3.] [3.]`.

I also checked scale conditioning: with the same seed and unchanged `mu`, doubling the stored `sigma` should double the new-group residual about each draw's `mu`. The maximum absolute residual error is `4.44e-16` in dense mode and `2.6165` in sparse mode; the known-group controls are exact. The location failure also reproduces with non-centered priors, but the centered example above avoids any dependency on #1003.

Environment: Python 3.13.14, PyMC 6.3.2, PyTensor 3.3.1, Formulae 0.7.0, NumPy 2.4.6, SciPy 1.17.1. Bambi was installed directly from the exact upstream commit above; its installed commit was verified, not an editable fork checkout.

## Likely cause and scope

In [`add_new_sparse_group_specific_contributions`](https://github.com/bambinos/bambi/blob/5b7e1b1f42097adeb1b9d846e30c31aa91616332/bambi/backend/pymc/parameters/conditional/prediction.py#L145-L156), the existing coefficient block is remapped through the cloned graph's `memo`, but the newly created coefficient expression is appended without that remapping. The [dense path](https://github.com/bambinos/bambi/blob/5b7e1b1f42097adeb1b9d846e30c31aa91616332/bambi/backend/pymc/parameters/conditional/prediction.py#L80-L86) remaps the complete expression.

A temporary in-memory insertion of `replace_vars_in_graphs([new_coefficients], memo)[0]` before flattening fixed the location-shift diagnostic on the #1066 branch. This is a candidate fix, not yet a fully tested patch; tests should cover both location and scale conditioning, aliases, slopes/multivariate shapes, and repeated new labels.

The demonstrated failure concerns prediction for new group labels. It does not establish a general fitting failure or a security vulnerability. Dense construction (`SPARSE_DOT=False` before building the model) works in this reproducer. This issue is separate from the non-centered Normal location correction in #1066.
